src/store/reducers/todo.js

```diff
@@ -16,6 +16,13 @@ export const todoReducer = (state = initialTodoState, action) => {
       };
     }
 
+    case DELETE_TASK: {
+      const existingTasks = structuredClone(state.tasks);
+      const selectedTaskId = findTaskIndexById(action.payload, existingTasks);
+      existingTasks.splice(selectedTaskId, 1);
+      return { ...state, tasks: existingTasks };
```

Comment:
@azo-wtag Bhai, any specific reason for doing it this way? Like: we could have just filtered the list and returned the updated one. Just curious about knowing the benefits here. 💡

Reply:
No specific reason for deleting item using this way. While solving this problem this process came to mind first thats why implemented in this way. @fza-wtag vai